Sources/FirebaseAuth/FirebaseAuthError.swift

```diff
@@ -1,117 +1,150 @@
+  public static var maximumSecondFactorCountExceeded: Self { .init(firebase.auth.kAuthErrorMaximumSecondFactorCountExceeded) }
+  public static var unsupportedFirstFactor: Self { .init(firebase.auth.kAuthErrorUnsupportedFirstFactor) }
+  public static var emailChangeNeedsVerification: Self { .init(firebase.auth.kAuthErrorEmailChangeNeedsVerification) }
   #if INTERNAL_EXPERIMENTAL
-  case invalidEventHandler = 79
-  case federatedProviderAlreadyInUse = 80
-  case invalidAuthenticatedUserData = 81
-  case federatedSignInUserInteractionFailure = 82
-  case missingOrInvalidNonce = 83
-  case userCancelled = 84
-  case unsupportedPassthroughOperation = 85
-  case tokenRefreshUnavailable = 86
+  public static var invalidEventHandler: Self { .init(firebase.auth.kAuthErrorInvalidEventHandler) }
+  public static var federatedProviderAlreadyInUse: Self { .init(firebase.auth.kAuthErrorFederatedProviderAlreadyInUse) }
+  public static var invalidAuthenticatedUserData: Self { .init(firebase.auth.kAuthErrorInvalidAuthenticatedUserData) }
+  public static var federatedSignInUserInteractionFailure: Self { .init(firebase.auth.kAuthErrorFederatedSignInUserInteractionFailure) }
+  public static var missingOrInvalidNonce: Self { .init(firebase.auth.kAuthErrorMissingOrInvalidNonce) }
+  public static var userCancelled: Self { .init(firebase.auth.kAuthErrorUserCancelled) }
+  public static var unsupportedPassthroughOperation: Self { .init(firebase.auth.kAuthErrorUnsupportedPassthroughOperation) }
+  public static var tokenRefreshUnavailable: Self { .init(firebase.auth.kAuthErrorTokenRefreshUnavailable) }
   #endif
 
   // Errors that are not represented in the C++ SDK, but are
   // present in the reference API.
-  case missingAndroidPackageName = 17037
-  case webNetworkRequestFailed = 17061
-  case webSignInUserInteractionFailure = 17063
-  case localPlayerNotAuthenticated = 17066
-  case nullUser = 17067
-  case gameKitNotLinked = 17076
-  case secondFactorRequired = 17078
-  case blockingCloudFunctionError = 17105
-  case internalError = 17999
-  case malformedJWT = 18000
+  public static var missingAndroidPackageName: Self { .init(rawValue: 17037) }
+  public static var webNetworkRequestFailed: Self { .init(rawValue: 17061) }
+  public static var webSignInUserInteractionFailure: Self { .init(rawValue: 17063) }
+  public static var localPlayerNotAuthenticated: Self { .init(rawValue: 17066) }
+  public static var nullUser: Self { .init(rawValue: 17067) }
+  public static var gameKitNotLinked: Self { .init(rawValue: 17076) }
+  public static var secondFactorRequired: Self { .init(rawValue: 17078) }
+  public static var blockingCloudFunctionError: Self { .init(rawValue: 17105) }
+  public static var internalError: Self { .init(rawValue: 17999) }
+  public static var malformedJWT: Self { .init(rawValue: 18000) }
 }
 
-extension AuthErrorCode: Error {}
+extension AuthErrorCode: Equatable {}
+
 extension AuthErrorCode {
+  public typealias Code = AuthErrorCode
```

Comment:
Is this in the reference implementation?

Reply:
`AuthErrorCode.Code` exists under Obj C, yeah. I imagine it is defined a bit differently. The `code` is `self` thing seems a bit hacky to me.

Reply:
Added comment.